### ISSUE TITLE
Kiwix JS on Edge Chromium

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#486]
[This issue is just for information.] Edge Chromium is now available for developer testing on Windows 10 from https://www.microsoftedgeinsider.com/en-us/ . It uses the Blink engine instead of EdgeHTML..

I have tested Kiwix JS and Kiwix JS Windows on it in various scenarios, and here are the results:

* To enable running Kiwix JS from the file system, Edge Chromium must be launched with the `--allow-file-access-from-files` switch;

* Like all the other browsers, Service Worker mode works only from localhost with a server, in an extension, or served over the Web, not when accessing the app from the file system;

* The current Kiwix JS extension can be installed from the Chrome Web Store (after allowing access to non-Microsoft extension sources), and Kiwix JS then runs as an extension with no perceivable issues in both jQuery and SW modes.

To summarize, Kiwix JS runs exaclty as it does in Chromium. It's a shame that access from the filesystem requires the startup flag, unlike Firefox and original Edge, but the fact that Kiwix JS runs as an extension does mitigate this usability issue.

My main suggestion from this is that we should in due course offer the extension code to the Microsoft Store as well as the Chrome Web Store, for maximum diffusion of the app and ease of installation. I'm happy to look into this.

**Comment by kelson42** [comment]
@Jaifroid thx for test. Looks promising. I of course support the additional support of Edge Chromium.

**Comment by mossroy** [comment]
Cool, that's what I was hoping. Thanks for this complete test!

Does the Microsoft store use the same extension format as Chrome? In other words, can we upload the same unsigned .crx file to the Microsoft store, or will we need to add extra steps in our Travis CI, to produce a different file?
You might test with the nightly builds from https://download.kiwix.org/nightly/

**Comment by Jaifroid** [comment]
> Does the Microsoft store use the same extension format as Chrome?

The fact that (reportedly) most extensions (and for sure Kiwix JS) can be installed directly from the Chrome store in Edge Chromium suggests that the format is exactly the same. However, I can't open the .crx file in https://download.kiwix.org/nightly/ -- the browser reports "Package is invalid: CRX_HEADER_INVALID". I haven't tried loading the unpackaged version of the extension yet, as I need to investigate the procedure.

**Comment by Jaifroid** [comment]
Update: by enabling Developer mode in Extensions settings, it is possible to load the unpacked Kiwix JS extension (i.e., unzip the .zip file to a folder and then select that folder on the Extensions page of the browser). This works fine, so it suggests we'll be able to use the same file format.

**Comment by Jaifroid** [comment]
Microsoft has now made Edge Chromium the default browser on Windows 10, and for users who do not know how to set a specific registry key, access to Edge Legacy (which remains available for webviews and UWP apps) has been disabled. We should maintain our Unite Tests for Edge Legacy because its engine is still used in UWP apps and I believe it is still the default browser in Windows 8.0 / 8.1. I don't think it's a priority to add a specific test for Edge Chromium, given that we test Chrome, and I test manually in Edge Chromium. In any case, I think this issue can be closed.

**Comment by mossroy** [comment]
@Jaifroid many thanks for all this cleanup.
I was planning to do that : you've been faster than me.
I agree with all the issues you closed
